File: BackEnd/src/domain/interfaces/usecases/plants/IDeletePlantUseCase.ts
/**
 * @file IDeletePlantUseCase.ts
 * @description Interfaz del caso de uso para eliminar (soft-delete) una planta.
 * @module Plants
 * @layer Domain
 */
export interface IDeletePlantUseCase {
  execute(plantId: string, userId: string): Promise<void>;
}

File: FrontEnd/lib/domain/interfaces/usecases/plants/i_delete_plant_use_case.dart
/// @file i_delete_plant_use_case.dart
/// @description Interfaz del caso de uso para eliminar una planta (soft-delete).
/// @module Plants
/// @layer Domain
library;
abstract interface class IDeletePlantUseCase {
  /// Elimina (soft-delete) la planta con [plantId].
  Future<void> execute(String plantId);
}
